plugin/session_view.py

```diff
@@ -267,5 +273,41 @@ def _start_progress_reporter_async(self, request_id: int, title: str) -> ViewPro
         self.progress[request_id] = progress
         return progress
 
+    # --- textDocument/codeLens ----------------------------------------------------------------------------------------
+
+    def start_code_lenses_async(self) -> None:
+        params = {'textDocument': text_document_identifier(self.view)}
+        for request_id, request in self.active_requests.items():
+            if request.method == "codeAction/resolve":
+                self.session.send_notification(Notification("$/cancelRequest", {"id": request_id}))
+        self.session.send_request_async(
+            Request("textDocument/codeLens", params, self.view),
+            self._on_code_lenses_async
+        )
+
+    def _on_code_lenses_async(self, response: Optional[List[CodeLens]]) -> None:
+        self._code_lenses.clear_annotations()
+        if not isinstance(response, list):
+            return
+        self._code_lenses.handle_response(self.session.config.name, response)
+        self.resolve_visible_code_lenses_async()
+
+    def resolve_visible_code_lenses_async(self) -> None:
+        if self._code_lenses.is_empty():
+            self.start_code_lenses_async()
+            return
+
+        promises = []  # type: List[Promise[None]]
+        for code_lens in self._code_lenses.unresolved_visible_code_lens(self.view.visible_region()):
+            callback = functools.partial(code_lens.resolve, self.view)
+            promise = self.session.send_request_task(
+                Request("codeLens/resolve", code_lens.data, self.view)
+            ).then(callback)
+            promises.append(promise)
+        Promise.all(promises).then(lambda _: self._code_lenses.render())
+
+    def get_resolved_code_lenses_for_region(self, region: sublime.Region) -> Generator[CodeLens, None, None]:
+        yield from self._code_lenses.get_resolved_code_lenses_for_region(region)
```

Comment:
This code is equivalent to the old one so not a new issue but I don't really see the point of returning a generator when the caller is always gonna iterate through all the results anyway. I haven't measured it but it must be quite a bit slower then just returning a plain or even a tuple.